samples/matter/lock/child_image/mcuboot/boards/nrf52840dk_nrf52840.overlay

```diff
@@ -0,0 +1,11 @@
+/*
```

Comment:
It probably should be done in a lot of places. I did it here hoping to fix the CHIP door lock tests that were failing, and it seems to have worked.

Basically, it needs to be done if using external flash as one of the slots for MCUboot.

Reply:
OK, but it used to not be needed somehow. Is it a regression then?

Reply:
It's basically been lucky. Except when the app is non-secure, then it has silently failed. I don't know the details about how it fails. It seems the only test that was failing was the door lock one, so maybe we can merge this, and I can open a PR that changes more samples. @tejlmand could you comment?

Reply:
My point is that if this test has failed because of your change then probably other samples will fail on nightly as well. In the integration test suite we may not cover all samples. So it's good to understand which apps must be updated and update them all.

Reply:
@tejlmand or should I instead create a `pm_ext_flash.overlay` and do:
```
  if (CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY)
    add_overlay_dts(
      mcuboot
      ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/pm_ext_flash.overlay
      )
  endif()

Reply:
@Damian-Nordic @tejlmand I added overlay files to a number of samples now

Reply:
Thanks!

Reply:
Don't we need also to align `matter/window_covering` and `applications/matter_weather_station`?